src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts

```diff
@@ -31,6 +33,9 @@ export class DatasetService {
     if (this.uiSettings.get(UI_SETTINGS.QUERY_ENHANCEMENTS_ENABLED)) {
       this.registerDefaultTypes();
     }
+    this.recentDatasets = new LRUCache({
+      max: this.uiSettings.get(UI_SETTINGS.SEARCH_MAX_RECENT_DATASETS) ?? 4,
```

Comment:
nit: could this actually be undefined?

Reply:
While running tests, `uiSettings` has everything set to be `jest.fn()`, which is the only place where it would be undefined. Could override the original `uiSettings` mock such that `uiSettings.get()` returns `4`, would that be better practice?

Reply:
yes, dont change the UI implementation to meet a test infrastructure gap if possible.